css-view-transitions-2/Overview.bs

```diff
@@ -0,0 +1,373 @@
+			);
+
+			await event.viewTransition.ready;
+
+			// Animate the new document's view
+			document.documentElement.animate(
+				{
+					clipPath: [
+						`circle(0 at ${x}px ${y}px)`,
+						`circle(${endRadius}px at ${x}px ${y}px)`,
+					],
+				},
+				{
+					duration: 500,
+					easing: 'ease-in',
+					pseudoElement: '::view-transition-new(root)'
+				}
+			);
+		})
+		```
+	</div>
+
+
+
+# CSS rules # {#css-rules}
+
+## The <dfn id="at-auto-view-transition-rule">''@auto-view-transition''</dfn> rule ## {#auto-view-transition-rule}
+<h3 id="syntax-page-selector">@auto-view-transition rule grammar</h3>
+
+''@auto-view-transition'' rules are [=CSS/parsed=] according to the following grammar,
+plus the additional rules noted below:
+
+<pre class=prod>
+	@auto-view-transition = @auto-view-transition { <<declaration-rule-list>> }
+</pre>
+
+## The [=@auto-view-transition/same-origin=] property ## {#view-transition-name-prop}
+
+	<pre class='descdef'>
+	Name: same-origin
+	For: @auto-view-transition
+	Value: enabled | disabled
+	Initial: disabled
+	</pre>
+
+	The '<dfn for="@auto-view-transition">same-origin</dfn>' property opts in to automatically performing a view transition when performing a [=same origin=] navigation.
+	It needs to be enabled both in the old document (when unloading) and in the new document (when ready to render).
+
+	<dl dfn-type=value dfn-for="same-origin">
+		: <dfn>disabled</dfn>
+		:: There will be no transition.
+
+		: <dfn>enabled</dfn>
+		:: The transition will be enabled if the navigation is same-origin, without cross-origin
+			redirects.
+	</dl>
+
+# API # {#api}
+
+## The <dfn interface>PageRevealEvent</dfn> ## {#reveal-event}
+
+Note: this should go in the HTML spec. See [Issue 9315](https://github.com/whatwg/html/issues/9315).
+
+<xmp class=idl>
+		[Exposed=Window]
+		interface PageRevealEvent : Event {
+			readonly attribute ViewTransition? viewTransition;
+		};
+</xmp>
+
+The <dfn attribute for=PageRevealEvent>viewTransition</dfn> [=getter steps=] are to return the
+[=inbound cross-document view-transition=] for [=this's=] [=relevant global object's=] [=associated document=].
+
+
+
+# Algorithms # {#algorithms}
+
+## Additions to {{ViewTransition}} ## {#view-transitions-extension}
+
+	A {{ViewTransition}} additionally has:
+	<dl dfn-for=ViewTransition>
+		: <dfn>is cross-document</dfn>
+		:: a boolean, initially false.
+	</dl>
+
+## Monkey patches to HTML ## {#monkey-patch-to-html}
+
+	<div algorithm="monkey patch to unload">
+		Run the following step before <a href="https://html.spec.whatwg.org/multipage/document-lifecycle.html#unloading-documents:update-the-visibility-state">updating the visibility state</a> in the [=unload=] steps:
+
+		1. If |newDocument| is given, then [=setup outbound cross-document view transition=] given |oldDocument| and |newDocument|.
+	</div>
+
+	<div algorithm="monkey patch to render blocking">
+		Run the following step at the end of [=unblock rendering=]:
```

Comment:
"unblock rendering" is per element. We need a different link here. There is also subtlety about whether this event fires after the "load" events on all resources which were render blocking.

Reply:
It's OK that it's called per element. Why do we need a different link?
Regarding the subtlety: right, it should be unblocked after the load event. This is already the case for styles, we should change the order for scripts so that this is called after the load/error events.

Reply:
Added a monkey patch to script processing. The reveal event will also be called after the load/error events (this is already true for styles)

Reply:
Oh yes, there was an early return in `reveal a document` that checked this and I removed it by mistake.